sdk/python/feast/registry.py

```diff
@@ -477,3 +479,73 @@ def _write_registry(self, registry_proto: RegistryProto):
         file_obj.seek(0)
         blob.upload_from_file(file_obj)
         return
+
+
+class AwsS3RegistryStore(RegistryStore):
+    def __init__(self, uri: str):
+        self._uri = urlparse(uri)
+        self._bucket = self._uri.hostname
+        self._key = self._uri.path.lstrip("/")
+        return
+
+    def get_registry_proto(self):
+        import boto3
+        import botocore
+
+        file_obj = TemporaryFile()
+        registry_proto = RegistryProto()
+        s3 = boto3.resource("s3")
+        try:
+            bucket = s3.Bucket(self._bucket)
+            s3.meta.client.head_bucket(Bucket=bucket.name)
+        except botocore.client.ClientError as e:
+            # If a client error is thrown, then check that it was a 404 error.
+            # If it was a 404 error, then the bucket does not exist.
+            error_code = int(e.response["Error"]["Code"])
+            if error_code == 404:
+                raise Exception(
```

Comment:
We should probably have a custom exception. See https://github.com/feast-dev/feast/blob/master/sdk/python/feast/errors.py

Reply:
Should we also add similar for GCP as well or it would be better to open a new ticket?

Reply:
New PR